File: May 19, 2026.py
def sleep_debt(hours_slept, target_hours):

    hours_needed = 0

    total = sum(n - target_hours for n in hours_slept)

    if total > target_hours:
        hours_needed = 0
        return hours_needed

    if total > 0:
        hours_needed = target_hours - total

    elif total < 0:
        hours_needed = target_hours + abs(total)

    return hours_needed

File: May 19, 2026.js
function sleepDebt(hoursSlept, targetHours) {

  let hoursNeeded = 0;
  let sum = hoursSlept.reduce((acc, n) => acc + (n - targetHours), 0);

  if (sum > targetHours) {
    hoursNeeded = 0;
    return hoursNeeded
  }
  if (sum > 0) {
    hoursNeeded = targetHours - sum;
  } else if (sum < 0) {
    hoursNeeded = targetHours + Math.abs(sum);
  }

  return hoursNeeded;
}